ast/compile.go

```diff
@@ -2024,7 +2025,7 @@ func (qc *queryCompiler) checkVoidCalls(_ *QueryContext, body Body) (Body, error
 }
 
 func (qc *queryCompiler) checkUndefinedFuncs(_ *QueryContext, body Body) (Body, error) {
-	if errs := checkUndefinedFuncs(body, qc.compiler.GetArity); len(errs) > 0 {
+	if errs := checkUndefinedFuncs(body, qc.compiler.GetArity, qc.rewritten); len(errs) > 0 {
```

Comment:
For the typechecker, we're passing both `qc.rewritten` _and_ `qc.compiler.RewrittenVars` to `rewriteVarsInRef`... I don't think we need that here, but I'm not completely sure why we do _that there_.